builtin/logical/database/path_rotate_credentials.go

```diff
@@ -94,6 +130,11 @@ func (b *databaseBackend) pathRotateCredentialsUpdate() framework.OperationFunc
 			return nil, err
 		}
 
+		// Delete the WAL entry after successfully rotating root credentials
+		if err := framework.DeleteWAL(ctx, req.Storage, walID); err != nil {
+			b.Logger().Warn("unable to delete WAL", "error", err, "WAL ID", walID)
+		}
+
 		// Close the plugin
 		db.closed = true
 		if err := db.Database.Close(); err != nil {
```

Comment:
This is interesting - totally unrelated to this PR, but it seems like the database connection wouldn't get closed if we errored out above. This closing stuff seems like it should be in a `defer` on line 78.

I respect that this is out of scope for this PR, and I'm not asking you to investigate and potentially fix it. Just commenting upon it so we can take a closer look and see if there's an impact, and address it aside from this PR.

Reply:
Good find! I'll change this as a part of this PR to use `defer` on line 78 as you've mentioned.

Reply:
Fixed in 8a4dc2a. I also ensured this didn't have side effects by running tests and doing manual testing of multiple scenarios with the WAL rollback.

Reply:
Thank you, that was really kind of you to fix it while you're here.